jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/spec/client/exceptions/ClientExceptionsIT.java

```diff
@@ -0,0 +1,231 @@
+                // when
+                () -> client.target(uriBuilderForStatus(Status.NOT_ACCEPTABLE, baseUrl)).request().get(String.class),
+                // then
+                isThrowing(NotAcceptableException.class)
+        );
+
+        // Verifies that an instance of NotSupportedException is thrown if the server returns the status code 415.
+        assertThat(
+                // when
+                () -> client.target(uriBuilderForStatus(Status.UNSUPPORTED_MEDIA_TYPE, baseUrl)).request().get(String.class),
+                // then
+                isThrowing(NotSupportedException.class)
+        );
+
+        // Verifies that an instance of ServerErrorException is thrown if the server returns a 5XX status code.
+        assertThat(
+                // when
+                () -> client.target(uriBuilderForStatusCode(567, baseUrl)).request().get(String.class),
+                // then
+                isThrowing(ServerErrorException.class)
+        );
+
+        // Verifies that an instance of InternalServerErrorException is thrown if the server returns the status code 500.
+        assertThat(
+                // when
+                () -> client.target(uriBuilderForStatus(Status.INTERNAL_SERVER_ERROR, baseUrl)).request().get(String.class),
+                // then
+                isThrowing(InternalServerErrorException.class)
+        );
+
+        // Verifies that an instance of ServiceUnavailableException is thrown if the server returns the status code 503.
+        assertThat(
+                // when
+                () -> client.target(uriBuilderForStatus(Status.SERVICE_UNAVAILABLE, baseUrl)).request().get(String.class),
+                // then
+                isThrowing(ServiceUnavailableException.class)
+        );
+    }
+
+    @BeforeAll
+    static void createClient() {
+        ClientExceptionsIT.client = ClientBuilder.newClient();
+    }
+
+    @AfterAll
+    static void disposeClient() {
+        ClientExceptionsIT.client.close();
+    }
+
+    @ApplicationPath("")
+    private static final class StatusApplication extends Application {
+
+        @Override
+        public Set<Class<?>> getClasses() {
+            return Set.of(StatusResource.class);
+        }
+
+        @Path("status")
+        public static class StatusResource {
+
+            @GET
+            @Path("{statusCode}")
+            public Response respondWithStatusCode(@PathParam("statusCode") final int statusCode) {
+                return Response.status(statusCode).build();
+            }
+        }
+    }
+
+    private static final UriBuilder uriBuilderForStatus(Status status, URL baseUrl) {
+        return uriBuilderForStatusCode(status.getStatusCode(), baseUrl);
+    }
+
+    private static final UriBuilder uriBuilderForStatusCode(int statusCode, URL baseUrl) {
+        try {
+            return UriBuilder.fromUri(baseUrl.toURI())
+                    .path("status")
+                    .path(Integer.toString(statusCode));
```

Comment:
I wonder if `.build(statusCode)` wouldn't be possibly instead, thanks to autoboxing?